internal/policy/policy.go

```diff
@@ -36,21 +36,54 @@ import (
 	"github.com/hacbs-contract/ec-cli/internal/kubernetes"
 )
 
-type Policy struct {
+type Policy interface {
+	PublicKeyPEM() ([]byte, error)
+	CheckOpts() (*cosign.CheckOpts, error)
+	WithSpec(spec ecc.EnterpriseContractPolicySpec) Policy
+	Spec() ecc.EnterpriseContractPolicySpec
+	EffectiveTime() time.Time
+}
+
+type policy struct {
 	ecc.EnterpriseContractPolicySpec
-	CheckOpts     *cosign.CheckOpts
-	EffectiveTime time.Time
+	checkOpts     *cosign.CheckOpts
+	effectiveTime time.Time
 }
 
 // PublicKeyPEM returns the PublicKey in PEM format.
-func (p *Policy) PublicKeyPEM() ([]byte, error) {
-	pk, err := p.CheckOpts.SigVerifier.PublicKey()
+func (p *policy) PublicKeyPEM() ([]byte, error) {
+	pk, err := p.checkOpts.SigVerifier.PublicKey()
 	if err != nil {
 		return []byte{}, err
 	}
 	return cryptoutils.MarshalPublicKeyToPEM(pk)
 }
 
+func (p *policy) CheckOpts() (*cosign.CheckOpts, error) {
+	if p.checkOpts == nil {
+		return nil, errors.New("no check options configured")
```

Comment:
This seems a bit unusual. Is returning an error the right thing here?

Reply:
Either that or a panic, `p.checkOpts` might not get set...